x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/LogicalPlanBuilder.java

```diff
@@ -285,6 +286,25 @@ public LogicalPlan visitFromCommand(EsqlBaseParser.FromCommandContext ctx) {
         );
     }
 
+    @Override
+    public PlanFactory visitInsistCommand(EsqlBaseParser.InsistCommandContext ctx) {
+        var source = source(ctx);
+        List<NamedExpression> fields = visitQualifiedNamePatterns(ctx.qualifiedNamePatterns(), ne -> {
+            Source neSource = ne.source();
+            if (ne instanceof UnresolvedStar) {
+                throw new ParsingException(neSource, "Cannot specify [*] with INSIST", neSource.text());
+            }
+            if (ne instanceof UnresolvedNamePattern) {
+                throw new ParsingException(neSource, "Cannot use wildcards ([*]) with INSIST", neSource.text());
+            }
```

Comment:
potentially use a string with a null value that gets initialized in one of the two ifs and if set, triggers a ParsingException.
Avoid the constructor repetition and extraction of ne's source.